### PR TITLE
test(backend): add and improve unit tests across core services and controllers

### PR DESCRIPTION
**Pull request by meetjain6091**
PR Checklist

 The PR title follows the commit guidelines.

 There is an issue for the feature this PR is for.

 All existing tests and lints pass.

What is the current behavior?

Several backend services and controllers had low or missing unit test coverage.
This made it difficult to confidently validate business logic, detect regressions, and maintain long-term code quality—especially across complex domains like inventory, invoices, tasks, organizations, and documents.

What is the new behavior?

This PR significantly improves backend reliability by adding and updating comprehensive unit tests across core services and controllers, including:

Inventory, invoice, task, service, and organization services

Observation tool submission and document-related services

Parent, parent-companion, Stripe, and user profile services

Dashboard, form, and controller-level test improvements

Key outcomes:

70/70 test suites passing

1503/1503 tests passing

Overall backend coverage improved to ~92% lines and ~94% functions

Better branch and edge-case validation across critical flows

These changes increase confidence in backend behavior and reduce the risk of regressions during future development.

Related Issue(s)

Fixes #

### CODE REVIEW

**Comment by aupyay** [comment]
codex

File: apps/backend/test/middlewares/upload.test.ts [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+  // ======================================================================
+  // 1. UTILITY FUNCTIONS & CONFIG
+  // ======================================================================
+  describe('Utilities & Configuration', () => {
+    it('should throw if bucket name env is missing', () => {
+      delete process.env.AWS_S3_BUCKET_NAME;
+      // We need to wrap a function that calls getBucketName internally.
+      // Since getBucketName is not exported directly, we verify via a public function that uses it.
+      expect(UploadMiddleware.generatePresignedUrl('image/png', 'user', '123'))
+        .rejects.toThrow('AWS_S3_BUCKET_NAME is not defined');
```
[the remark this answers is not in the document]
Comment (meetjain6091) [reply apps/backend/test/middlewares/upload.test.ts]: 👍

**Comment by harshvardhan-yc** [comment]
codex

**Review by harshvardhan-yc — APPROVED** [review]
LGTM
